File: src/romi_soccer/nodes/motor_control.cpp
#include <ros/ros.h>
#include "geometry_msgs/Twist.h"
#include <string>
#include <Romi32U4Encoders.h>


// Noah, use this or the Python one to convert a linear/angular velocity command to an output for the romi using the encoders
void callback(const geometry_msgs::Twist::ConstPtr &msg){
     if (Romi32U4Encoders::getCountsLeft == )
}


int main(int argc, char **argv){
     ros::init(argc, argv, 'motor_control');
     ros::NodeHandle nh;
     std::string team_name;
     std::string shape_name;
     nh.getParam('team', team_name);
     nh.getParam('shape', shape_name);
     ros::Subscriber sub = nh.subscribe('%s_team/%s/cmd_vel' % team_name % shape_name, 1000, &callback);
     ros::spin()
}

File: src/romi_soccer/nodes/motor_control.py
#!/usr/bin/env python
import rospy
from geometry_msgs.msg import Twist

# Convert linear/angular velocity commands on cmd_vel topic to the right output for the encoders/motor pins or whatever
def callback(msg,team_name,shape_name):


if __name__=='__main__':
    # Initialize node to 'motor_control' (note: name can be overwritten in launch file)
    rospy.init_node('motor_control')
    # Get parameters for team_name and shape_name. I'm doing this so we can just call the same file to use for each node
    team_name = rospy.get_param('~team')
    shape_name = rospy.get_param('~shape')
    # Subscribe to the cmd_vel topic of the specific robot defined in the parameters
    rospy.Subscriber('%s/%s/cmd_vel' %team_name %shape_name,Twist,callback,team_name,shape_name)
